appserver/static/visualizations/hman/src/buildDynamicOption/index.js

```diff
@@ -0,0 +1,227 @@
+      for (var i = 0; i < data.rows.length; i++) {
+        if (isNaN(xAxisDataIndex[j])) {
+          throw "Wrong configuration of 'xAxisDataIndexBinding'. Please provide a number or a comma seperated list of numbers. 'xAxisDataIndexBinding': " + configXAxisDataIndexBinding;
+        } else {
+          xAxisObjects[j].data.push(data.rows[i][xAxisDataIndex[j]]);
+        }
+      }
+    }
+  }
+  if (configXAxisDataIndexBinding != null) {
+    if (option.hasOwnProperty('xAxis')) {
+      option['xAxis'] = xAxisObjects;
+    } else {
+      option.xAxis = xAxisObjects;
+    }
+  }
+
+  // Default settings possibly defined in Splunk's dynamicOptions
+  var defaultName = option.dynamicOptions.defaultName || 'Series ';
+  var defaultType = option.dynamicOptions.defaultType || 'line';
+  var defaultSmooth = option.dynamicOptions.defaultSmooth || false;
+  var defaultConnectNulls = option.dynamicOptions.defaultConnectNulls || true;
+  var defaultShowSymbol = option.dynamicOptions.defaultShowSymbol || false;
+
+  var onlyDrawSpecified = option.dynamicOptions.onlyDrawSpecified || false;
+
+  const seriesList = [];
+
+  // A helper function to create series data
+  function createSeries(seriesName, seriesType, seriesDataIndex, seriesSmooth, seriesConnectNulls, seriesShowSymbol) {
+    var data_array = data.rows.map(function (row) {
+      return [row[0], row[seriesDataIndex]];
+    });
+    return {
+      name: seriesName,
+      type: seriesType,
+      smooth: seriesSmooth,
+      connectNulls: seriesConnectNulls,
+      showSymbol: seriesShowSymbol,
+      data: data_array
+    };
+  }
+
+  // Generate series that are explicitly specified
+  for (var i = 0; i < seriesDataIndexes.length; i++) {
+    var seriesName = option.series && option.series[i] && option.series[i].name ? option.series[i].name : defaultName + (i + 1);
+    var seriesType = option.series && option.series[i] && option.series[i].type ? option.series[i].type : defaultType;
+    var seriesSmooth = option.series && option.series[i] && option.series[i].smooth ? option.series[i].smooth : defaultSmooth;
+    var seriesConnectNulls = option.series && option.series[i] && option.series[i].connectNulls ? option.series[i].connectNulls : defaultConnectNulls;
+    var seriesShowSymbol = option.series && option.series[i] && option.series[i].showSymbol ? option.series[i].showSymbol : defaultShowSymbol;
+
+    seriesList.push(createSeries(seriesName, seriesType, seriesDataIndexes[i], seriesSmooth, seriesConnectNulls, seriesShowSymbol));
+  }
+
+  // If not onlyDrawSpecified, generate all series with default settings
+  if (!onlyDrawSpecified) {
+    for (var i = 1; i < data.fields.length; i++) {
+      // If the series is already specified, skip it
+      if (seriesDataIndexes.indexOf(i) !== -1) continue;
+
+      seriesList.push(createSeries(defaultName + i, defaultType, i, defaultSmooth, defaultConnectNulls, defaultShowSymbol));
+    }
+  }
+
+  option.series = seriesList;
+
```

Comment:
following error handling up to the end of the file is identical to _buildCustomOption